src/dmtest/vdo/vdo_stack.py

```diff
@@ -0,0 +1,46 @@
+import dmtest.device_mapper.dev as dmdev
+import dmtest.device_mapper.table as table
+import dmtest.device_mapper.targets as targets
+
+from dmtest.process import run
+
+
+class VDOStack:
+    def __init__(self, data_dev, **opts):
+        self._data_dev = data_dev
+	self._physical_blocks = opts.pop("physical_blocks", utils.dev_size(data_dev) / 4096)
+	self._mode = opts.pop("512_mode", False)
+	self._block_map_cache = opts.pop("block_map_cache", 128 * 1024 * 1024)
+	self._block_map_period = opts.pop("block_map_period", 16380)
+        self._format = opts.pop("format", True)
+        self._logical_size = opts.pop("logical_size", "20G")
+        self._alb_mem = opts.pop("albireo_mem", 0.25)
+        self._alb_sparse = opts.pop("albireo_sparse", False)
+        self._opts = opts
+
+        if self._format:
+            logical_size = "--logical-size=" + self._logical_size
+            mem = "--uds-memory-size=" + self._alb_mem
+            sparse = ""
+            if self._alb_sparse:
+                sparse = " --uds-sparse"
+            dev = self._data_dev
+            run(f"vdoformat --force {logical_size} {mem}{sparse} {dev}")
+
+
+    def _vdo_table(self):
+        return table.Table(
+            targets.VDOTarget(
+                self._data_dev,
```

Comment:
A parameter is missing. The first parameter should be sector_count, based on the input args expected in targets.VDOTarget().